x/signer/client/cli/suite_test.go

```diff
@@ -0,0 +1,378 @@
+	s.Require().NoError(err)
+	s.kr = kr
+	s.baseCtx = client.Context{}.
+		WithKeyring(s.kr).
+		WithTxConfig(s.encCfg.TxConfig).
+		WithCodec(s.encCfg.Codec).
+		WithClient(clitestutil.MockCometRPC{Client: rpcclientmock.Client{}}).
+		WithAccountRetriever(client.MockAccountRetriever{}).
+		WithOutput(io.Discard).
+		WithChainID("test-chain")
+
+	ctxGen := func() client.Context {
+		bz, _ := s.encCfg.Codec.Marshal(&sdk.TxResponse{})
+		c := clitestutil.NewMockCometRPC(abci.ResponseQuery{
+			Value: bz,
+		})
+		return s.baseCtx.WithClient(c)
+	}
+	s.clientCtx = ctxGen()
+
+	kb := s.clientCtx.Keyring
+	valAcc, _, err := kb.NewMnemonic("newAccount", keyring.English, sdk.FullFundraiserPath, keyring.DefaultBIP39Passphrase, hd.Secp256k1)
+	s.Require().NoError(err)
+	s.val, err = valAcc.GetAddress()
+	s.Require().NoError(err)
+
+	account1, _, err := kb.NewMnemonic("newAccount1", keyring.English, sdk.FullFundraiserPath, keyring.DefaultBIP39Passphrase, hd.Secp256k1)
+	s.Require().NoError(err)
+	s.val1, err = account1.GetAddress()
+	s.Require().NoError(err)
+
+	account2, _, err := kb.NewMnemonic("newAccount2", keyring.English, sdk.FullFundraiserPath, keyring.DefaultBIP39Passphrase, hd.Secp256k1)
+	s.Require().NoError(err)
+	pub1, err := account1.GetPubKey()
+	s.Require().NoError(err)
+	pub2, err := account2.GetPubKey()
+	s.Require().NoError(err)
+
+	// Create a dummy account for testing purpose
+	_, _, err = kb.NewMnemonic("dummyAccount", keyring.English, sdk.FullFundraiserPath, keyring.DefaultBIP39Passphrase, hd.Secp256k1)
+	s.Require().NoError(err)
+
+	multi := kmultisig.NewLegacyAminoPubKey(2, []cryptotypes.PubKey{pub1, pub2})
+	_, err = kb.SaveMultisig("multi", multi)
+	s.Require().NoError(err)
+
+	s.ac = addresscodec.NewBech32Codec("cosmos")
+}
+
+func (s *CLITestSuite) TestCLIValidateSignatures() {
+	sendTokens := sdk.NewCoins(
+		sdk.NewCoin("testtoken", math.NewInt(10)),
+		sdk.NewCoin("stake", math.NewInt(10)))
+
+	res, err := s.createBankMsg(s.clientCtx, s.val, sendTokens,
+		fmt.Sprintf("--%s=true", flags.FlagGenerateOnly))
+	s.Require().NoError(err)
+
+	// write  unsigned tx to file
+	unsignedTx := testutil.WriteToNewTempFile(s.T(), res.String())
+	defer unsignedTx.Close()
+
+	res, err = TxSignExec(s.clientCtx, s.val, s.keyringDir, unsignedTx.Name())
+	s.Require().NoError(err)
+	signedTx, err := s.clientCtx.TxConfig.TxJSONDecoder()(res.Bytes())
+	s.Require().NoError(err)
+
+	// write signed tx to file
+	signedTxFile := testutil.WriteToNewTempFile(s.T(), res.String())
+	defer signedTxFile.Close()
+
+	// verify signed tx
+	// txBuilder, err := s.clientCtx.TxConfig.WrapTxBuilder(signedTx)
+	_, err = s.clientCtx.TxConfig.WrapTxBuilder(signedTx)
+	s.Require().NoError(err)
+	_, err = TxValidateSignaturesExec(s.clientCtx, s.keyringDir, signedTxFile.Name())
+	s.Require().NoError(err)
+
+	// TODO: modifying the tx and verifying isn't working. It always passes when it should fail...
```

Comment:
@giunatale modifying the transaction should result in an error but is not. I assume this is because local verification doesn't check tx data and needs to be connected to a node?